src/lambda/config/templates.ts

```diff
@@ -93,47 +99,178 @@ export async function load(
+function getParseErrorDescription(code: jsonParser.ParseErrorCode): string {
     switch (code) {
-        case ParseErrorCode.CloseBraceExpected:
+        case jsonParser.ParseErrorCode.CloseBraceExpected:
             return 'close brace expected'
-        case ParseErrorCode.CloseBracketExpected:
+        case jsonParser.ParseErrorCode.CloseBracketExpected:
             return 'close bracket expected'
-        case ParseErrorCode.ColonExpected:
+        case jsonParser.ParseErrorCode.ColonExpected:
             return 'colon expected'
-        case ParseErrorCode.CommaExpected:
+        case jsonParser.ParseErrorCode.CommaExpected:
             return 'command expected'
-        case ParseErrorCode.EndOfFileExpected:
+        case jsonParser.ParseErrorCode.EndOfFileExpected:
             return 'end of file expected'
-        case ParseErrorCode.InvalidCharacter:
+        case jsonParser.ParseErrorCode.InvalidCharacter:
             return 'invalid character'
-        case ParseErrorCode.InvalidCommentToken:
+        case jsonParser.ParseErrorCode.InvalidCommentToken:
             return 'invalid comment token'
-        case ParseErrorCode.InvalidEscapeCharacter:
+        case jsonParser.ParseErrorCode.InvalidEscapeCharacter:
             return 'invalid escape character'
-        case ParseErrorCode.InvalidNumberFormat:
+        case jsonParser.ParseErrorCode.InvalidNumberFormat:
             return 'invalid number format'
-        case ParseErrorCode.InvalidSymbol:
+        case jsonParser.ParseErrorCode.InvalidSymbol:
             return 'invalid symbol'
-        case ParseErrorCode.InvalidUnicode:
+        case jsonParser.ParseErrorCode.InvalidUnicode:
             return 'invalid unicode'
-        case ParseErrorCode.PropertyNameExpected:
+        case jsonParser.ParseErrorCode.PropertyNameExpected:
             return 'property name expected'
-        case ParseErrorCode.UnexpectedEndOfComment:
+        case jsonParser.ParseErrorCode.UnexpectedEndOfComment:
             return 'unexpected end of comment'
-        case ParseErrorCode.UnexpectedEndOfNumber:
+        case jsonParser.ParseErrorCode.UnexpectedEndOfNumber:
             return 'unexpected end of number'
-        case ParseErrorCode.UnexpectedEndOfString:
+        case jsonParser.ParseErrorCode.UnexpectedEndOfString:
             return 'unexpected end of string'
-        case ParseErrorCode.ValueExpected:
+        case jsonParser.ParseErrorCode.ValueExpected:
             return 'value expected'
         // By omitting the default case, we force the compiler to yell at us
         // if any enum members are added/removed/changed.
     }
 }
+
+export class TemplatesConfigPopulator {
+    private isDirty: boolean = false
+
+    public constructor(
+        private json: string,
+        private readonly modificationOptions: jsonParser.ModificationOptions = {
+            formattingOptions: {
+                insertSpaces: true,
+                tabSize: 4,
```

Comment:
Check out `getTabSize` from `configureLocalLambdas.ts`.

Reply:
The callers provide information from the editor

Reply:
If there is no editor open, we should still at least do this much:

```typescript
tabSize: new DefaultSettingsConfiguration('editor').readSetting<number>('tabSize') || 4
```